sdk/internal/log/log.go

```diff
@@ -12,32 +12,13 @@ import (
 	"time"
 )
 
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// NOTE: The following are exported as public surface area from azcore.  DO NOT MODIFY
```

Comment:
Why doesn't this stuff live in `azcore`?

Reply:
Logging is a very oddball, and hopefully rare, case.  It's a mix of public and internal surface area.  We only want customers to interact with a small subset of the APIs (the ones in this comment block).  The rest are intended for SDK authors so they can't go in `azcore`.

Reply:
We could define `Event` in `azcore` and use `string` in `internal`, we'd just have to do a little casting (not terrible, just a little ugly).  However, we'll still need a comment block indicating that some of these APIs are public surface area.

Reply:
Thinking a bit more about that, I'm not in favor of it as SDK authors would also have to perform casting.